core-tests/shared/src/test/scala/zio/ScheduleSpec.scala

```diff
@@ -579,13 +577,28 @@ object ScheduleSpec extends ZIOBaseSpec {
       ),
       suite("repetitions")(
         test("count")(checkRepetitions(Schedule.count)),
+        test("dayOfMonth")(checkRepetitions(Schedule.dayOfMonth(1))),
+        test("dayOfWeek")(checkRepetitions(Schedule.dayOfWeek(1))),
         test("fixed")(checkRepetitions(Schedule.fixed(1.second))),
         test("forever")(checkRepetitions(Schedule.forever)),
+        test("hourOfDay")(checkRepetitions(Schedule.hourOfDay(1))),
+        test("minuteOfHour")(checkRepetitions(Schedule.minuteOfHour(1))),
         test("recurs")(checkRepetitions(Schedule.forever)),
+        test("secondOfMinute")(checkRepetitions(Schedule.secondOfMinute(1))),
         test("spaced")(checkRepetitions(Schedule.spaced(1.second))),
         test("windowed")(checkRepetitions(Schedule.windowed(1.second)))
       )
-    )
+    ),
+    test("intersection of schedules recurring in bounded intervals") {
```

Comment:
This is good. It would be a good hackathon ticket to add more tests for intersection and union with the bounded intervals case.